### PR TITLE
Monorepo Summer Cleaning!

### PR DESCRIPTION
**Pull request by LightBounded**
### TL;DR
- I fixed shit
- If CI is failing, don't approve
- Pwetty pwease let me merge :3

### This pull request...
- Updates all our tooling: eslint, tsconfig, prettier and tailwind
  - Mainly on the eslint side
  - Stricter lint rules
- Stores a user's Clerk ID to the users table
- Renames club-site -> club
- Deletes the socials-bot
  - This bot needs to be completely rewritten using a wrapper library around the Discord API like Discord.js 
  - It was a pain in the ass having to use the Discord API directly 
- Fixes new lint and type errors introduced from updating our tooling
  - You don't want to know how many...
- Makes the db, validators and api package a compiled packages vs. JIT package 
  - This gives us more flexibility on how we can deploy our apps that use these packages. 
  - See more [here](https://turbo.build/repo/docs/core-concepts/internal-packages#compiled-packages)

### Next steps...
- Since we updated the way we store users, we need to check if user creation works
- Fix the way we update user profiles (this is broken because of these changes)
- Add MLH checkboxes

### CODE REVIEW

**Review by gitSean23 — APPROVED** [review]
My glorious king Caleb! What an amazing PR you have conceived, exquisite even! Oh dear heavens, I will not keep you any longer and will let you be on your way to code for eons more!
